Foundation/Host.swift

```diff
@@ -33,6 +33,18 @@ open class Host: NSObject {
     }
     
     static internal func currentHostName() -> String {
+#if os(Windows)
+      var dwLength: DWORD = 0
+      GetComputerNameExA(ComputerNameDnsHostname, nil, &dwLength)
+
+      let hostname: UnsafeMutablePointer<Int8>? =
+          UnsafeMutableBufferPointer<Int8>
+              .allocate(capacity: Int(dwLength + 1))
+              .baseAddress
+      defer { hostname?.deallocate() }
+      GetComputerNameExA(ComputerNameDnsHostname, hostname, &dwLength)
+      return String(cString: hostname!)
```

Comment:
Can a check be added to return `"localhost"` if `hostname` is empty?

Reply:
I don't think that is possible (that is, `hostname` being empty).  We know the size of the host name due to the query above, so we should have plenty of space.  I suppose that it is possible to hit the race condition with the TOCTOU here ... but that does feel like an edge case.

Reply:
I think its worth adding something like
```
guard dwLength > 0 && GetComputerNameExA(..) else {
  return "localhost"
}
```
just to cover the edge case for completeness, even if its unlikely to get used.